src/main/java/uk/gov/companieshouse/customerfeedbackapi/service/CustomerFeedbackService.java

```diff
@@ -0,0 +1,106 @@
+  private final CustomerFeedbackRepository customerFeedbackRepository;
+
+  @Value("${send-email-flag}")
+  private boolean emailSendFlag;
+
+  @Value("${customer-feedback-email}")
+  private String customerFeedbackEmail;
+
+  @Value("${kafka-api-endpoint}")
+  private String kafkaApiEndpoint;
+
+  @Value("${app-id}")
+  private String appId;
+
+  @Autowired
+  public CustomerFeedbackService(
+      CustomerFeedbackMapper customerFeedbackMapper,
+      CustomerFeedbackRepository customerFeedbackRepository) {
+    this.customerFeedbackMapper = customerFeedbackMapper;
+    this.customerFeedbackRepository = customerFeedbackRepository;
+  }
+
+  public void createCustomerFeedback(CustomerFeedbackDTO customerFeedbackDTO, String requestId)
+      throws SendEmailException {
+
+    ApiLogger.debugContext(requestId, "Processing customer feedback");
+
+    var customerFeedbackDAO = customerFeedbackMapper.dtoToDao(customerFeedbackDTO);
+
+    customerFeedbackDAO.setCreatedAt(LocalDateTime.now());
+
+    boolean emailSent = emailSendFlag && !customerFeedbackDAO.getData().getSourceUrl().isBlank();
+    customerFeedbackDAO.setEmailSent(emailSent);
+
+    ApiLogger.debugContext(requestId, "Inserting customer feedback record");
+    CustomerFeedbackDAO createdCustomerFeedback =
+        customerFeedbackRepository.insert(customerFeedbackDAO);
+
+    if (emailSent) {
+      JSONObject json_data = new JSONObject();
+      json_data.put("customer_feedback", customerFeedbackDTO.getCustomerFeedback());
+      json_data.put("customer_name", customerFeedbackDTO.getCustomerName());
+      json_data.put("customer_email", customerFeedbackDTO.getCustomerEmail());
+      json_data.put("source_url", customerFeedbackDTO.getSourceUrl());
+      json_data.put("kind", customerFeedbackDTO.getKind());
+      json_data.put("to", customerFeedbackEmail);
+      json_data.put("date", JSONObject.NULL);
+      String requestBody =
+          "app_id="
+              + appId
+              + "&"
+              + "message_id="
+              + UUID.randomUUID().toString()
+              + "&"
+              + "message_type="
+              + "customer-feedback"
+              + "&"
+              + "json_data="
+              + json_data.toString();
+      HttpURLConnection connection = null;
+      ApiLogger.debugContext(requestId, "Calling send-email endpoint");
+      try {
+        connection = (HttpURLConnection) new URL(kafkaApiEndpoint).openConnection();
+        connection.setRequestMethod("POST");
+        connection.setDoOutput(true);
+        connection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+        try (DataOutputStream out = new DataOutputStream(connection.getOutputStream())) {
+          out.write(requestBody.getBytes(StandardCharsets.UTF_8));
+          out.flush();
+        }
+        int responseCode = connection.getResponseCode();
+        ApiLogger.debugContext(requestId, "Response code from endpoint: " + responseCode);
```

Comment:
Could any post requests be built into the existing functionality in private-api-sdk-java?

Create a spec in private.api.ch.gov.uk-specifications
Use the latest version of the spec in private-api-sdk-java to create your new model in generated-sources
Create the functionality to send a post request for that model in private-api-sdk-java (see ChangedResource post for example to kafka-api)

Reply:
I think this should probably be a new ticket

Reply:
Primarily because I don't want to delay the delivery of the api. The estimate was three sprints. One is left for testing / pipeline / fixes. People are chasing up delivery. A separate ticket could stil be started while testing is being done

Reply:
Happy with that with the proviso that a tech debt ticket is created and worked on

Reply:
I'll create a ticket for it now